gradoop-data-integration/src/main/java/org/gradoop/dataintegration/importer/csv/MinimalCSVImporter.java

```diff
@@ -0,0 +1,169 @@
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config) {
+    this(path, tokenSeperator, config, "UTF-8");
+  }
+
+  /**
+   * Create a new MinimalCSVImporter with a user set charset. The first line of the file
+   * will set as the property names for each column.
+   * @param path the path to the csv file
+   * @param tokenSeperator the token delimiter of the csv file
+   * @param config GradoopFlinkConfig
+   * @param charset the charset used in the csv file
+   */
+  public MinimalCSVImporter(String path, String tokenSeperator, GradoopFlinkConfig config,
+          String charset) {
+    this.path = path;
+    this.tokenSeparator = tokenSeperator;
+    this.config = config;
+    this.charset = charset;
+  }
+
+  /**
+   * Import each row of the file as a vertex. If no column property names are set,
+   * read the first line of the file as header and set this values as column names.
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return the imported vertices
+ * @throws IOException if an error occurred while open the stream
+   */
+  public DataSet<ImportVertex<Long>> importVertices(boolean checkReoccurringHeader)
+          throws IOException {
+    if (columnNames == null) {
+      return readCSVFile(readHeaderRow(), checkReoccurringHeader);
+    } else {
+      return readCSVFile(columnNames, checkReoccurringHeader);
+    }
+  }
+
+  /**
+   * Read the vertices from a csv file.
+   *
+   * @param propertyNames list of the property identifier name
+   * @param checkReoccurringHeader if each row of the file should be checked for reocurring of
+   * the column property names.
+   * @return DateSet of all vertices from one specific file.
+   */
+  public DataSet<ImportVertex<Long>> readCSVFile(List<String> propertyNames,
+          boolean checkReoccurringHeader) {
+
+    DataSet<Properties> lines = config.getExecutionEnvironment()
+    .readTextFile(path)
+    .map(new RowToVertexMapper(path, tokenSeparator, propertyNames, checkReoccurringHeader))
+    .filter(new FilterNullValuesTuple<>());
+
+    return DataSetUtils.zipWithUniqueId(lines).map(new CreateImportVertexCSV<>());
+  }
+
+  /**
+   * Read the fist row of a csv file and put each the entry in each column in a list.
+   * @return the property names
+ * @throws IOException if an error occurred while open the stream
+   */
+  public List<String> readHeaderRow() throws IOException {
+    try (final BufferedReader reader =
+              new BufferedReader(new InputStreamReader(new FileInputStream(path),
+                      charset))) {
+      String headerLine = reader.readLine();
+      if (headerLine == null) {
+        LOG.error("The file do not contain any rows.");
+        throw new NullPointerException();
```

Comment:
I think it is not correct to throw a NullPointer here. I think you got me wrong earlier. If the method readHeaderRow would return null instead of an instantiated List<String> this would result into a NullPointerException elsewhere. Therefore we have to check that this doesn't happen. 
Indeed it is correct to throw an exception at this point but it should be a IoException which reads as follows:
`throw new IOException("The file " + path + " does not contain any rows.")`